pinot-common/src/main/java/com/linkedin/pinot/common/utils/EqualityUtils.java

```diff
@@ -17,125 +17,112 @@
 
 import java.util.ArrayList;
 import java.util.Arrays;
-import java.util.Collection;
 import java.util.Collections;
 import java.util.List;
-import java.util.Map;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
 
 
 /**
  * Various utilities to be used in implementing equals and hashCode.
  *
+ * Keep isEqual() for primitive type for historical reason, but try not using them.
```

Comment:
For me, I think '==' is more readable for primitive type, but this is personal preference. The main rationale is that the method call overhead is too expensive compare to the cost of primitive type comparison.